hnn/qt_thread.py

```diff
@@ -288,11 +240,14 @@ def run(self, sim_length=None):
         self.quit()
 
     def _run(self, banner=True, sim_length=None):
-        self.killed = False
+        with self.killed_lock:
+            self.killed = False
 
+        # make copy of params dict in Params object before
+        # modifying tstop
         sim_params = hnn_core_compat_params(self.params)
         if sim_length is not None:
-            sim_params['tstop'] = sim_length
+            sim_params['tstop'] = round(sim_length, 8)
```

Comment:
It's not clear to me why the rounding is necessary? I saw in the old commit that it is to avoid "precision errors"

Reply:
yeah hnn-core fails if tstop is 170.0000000002